lib/dal/src/migrations/U0035__components.sql

```diff
@@ -62,6 +75,309 @@ BEGIN
+            ident_nil_v1(), -- Prop ID
+            this_internal_provider.id, -- Internal Provider ID
+            ident_nil_v1(), -- External Provider ID
+            this_new_row.id -- Component ID
+        );
+
+        SELECT *
+        INTO STRICT this_attribute_prototype
+        FROM attribute_prototypes_v1(this_tenancy, this_visibility) AS ap
+        WHERE in_attribute_context_v1(this_attribute_context, ap);
+
+        SELECT av.object
+        INTO this_new_attribute_value
+        FROM attribute_value_create_v1(
+            this_tenancy,
+            this_visibility,
+            this_attribute_context,
+            this_unset_func_binding_id,
+            this_unset_func_binding_return_value_id,
+            NULL
+        ) AS av;
+
+        PERFORM set_belongs_to_v1(
+            'attribute_value_belongs_to_attribute_prototype',
+            this_tenancy,
+            this_visibility,
+            this_new_attribute_value ->> 'id',
+            this_attribute_prototype.id
+        );
+    END LOOP;
+
+    -- Implicit Internal Providers
+    FOR this_internal_provider IN
+        SELECT ip.*
+        FROM internal_providers_v1(this_tenancy, this_visibility) AS ip
+            INNER JOIN props_v1(this_tenancy, this_visibility) AS props
+              ON ip.prop_id = props.id
+        WHERE props.schema_variant_id = this_schema_variant_id
+    LOOP
+        -- Create an Attribute Value for the Internal Provider
+        this_attribute_context := attribute_context_build_from_parts_v1(
+            ident_nil_v1(), -- Prop ID
+            this_internal_provider.id, -- Internal Provider ID
+            ident_nil_v1(), -- External Provider ID
+            this_new_row.id -- Component ID
+        );
+
+        SELECT *
+        INTO STRICT this_attribute_prototype
+        FROM attribute_prototypes_v1(this_tenancy, this_visibility) AS ap
+        WHERE in_attribute_context_v1(this_attribute_context, ap);
+
+        SELECT av.object
+        INTO this_new_attribute_value
+        FROM attribute_value_create_v1(
+            this_tenancy,
+            this_visibility,
+            this_attribute_context,
+            this_unset_func_binding_id,
+            this_unset_func_binding_return_value_id,
+            NULL
+        ) AS av;
+
+        PERFORM set_belongs_to_v1(
+            'attribute_value_belongs_to_attribute_prototype',
+            this_tenancy,
+            this_visibility,
+            this_new_attribute_value ->> 'id',
+            this_attribute_prototype.id
+        );
+
+        -- Create an Attribute Value for the Prop.
+        this_attribute_context := attribute_context_build_from_parts_v1(
+            this_internal_provider.prop_id, -- Prop ID
+            ident_nil_v1(), -- Internal Provider ID
+            ident_nil_v1(), -- External Provider ID
+            this_new_row.id -- Component ID
+        );
+
+        SELECT *
+        INTO this_attribute_prototype
```

Comment:
I changed this from a INTO STRICT because I was observing multiple prototypes for a prop, even on main.

This may also be true for the other branches, for providers, but I don't know enough about this logic to say much. We should take a look into it.